components/brave_ads/browser/notification_helper_android.cc

```diff
@@ -0,0 +1,49 @@
+/* Copyright 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_ads/browser/notification_helper_android.h"
+#include "jni/NotificationSystemStatusUtil_jni.h"
+#include "jni/BraveAds_jni.h"
+#include "base/android/jni_string.h"
+
+namespace brave_ads {
+
+NotificationHelperAndroid::NotificationHelperAndroid() = default;
+
+NotificationHelperAndroid::~NotificationHelperAndroid() {
+  channels_provider_.reset();
```

Comment:
No need to reset as it is a smart pointer and will be destroyed when out of scope